src/quicklook.cs

```diff
@@ -176,11 +181,14 @@ interface QLPreviewItem {
 #endif
 	}
 
+	[NoMac]
 	delegate bool QLPreviewReplyDrawingHandler (CGContext context, QLPreviewReply reply, out NSError error);
+	[NoMac]
 	delegate NSData QLPreviewReplyDataCreationHandler (QLPreviewReply reply, out NSError error);
+	[NoMac]
 	delegate CGPDFDocument QLPreviewReplyUIDocumentCreationHandler (QLPreviewReply reply, out NSError error);
 
-	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[NoWatch, NoTV, NoMac, iOS (15,0), MacCatalyst (15,0)]
```

Comment:
This is both right and wrong at the same time.

It's right in the sense that you're removing `#if !MONOMAC`, and then to get the same result you need to add this attribute.
It's wrong because these APIs seem to actually exist in macOS (12).

I'm guessing we added these APIs at some point, with the correct availability attributes, not realizing that the code was inside a `#if !MONOMAC` block, and as such the API wouldn't be exposed on macOS.

The correct fix here is to maintain the attributes as-is (and remove the `#if MONOMAC` condition as you do), and then verify with intro/xtro/documentation that these APIs actually exist and are correct on macOS.

That's much bigger than just a mechanical conversion of this framework, but I think it's worth it (it's also a pretty small framework, so it shouldn't be too much work).

Reply:
@rolfbjarne So I just tried to apply the mac attributes to this one and a few of them below that say they are available for mac as well and they were getting a lot of compile errors.

It turns out that these are the questionable interfaces in this src/quicklook.cs 
QLPreviewReply
QLPreviewReplyAttachment
QLFilePreviewRequest
QLPreviewProvider

It turns out that they are also in src/quicklookui.cs and are all decorated with mac attributes already.
Do you think this could be leading to the issue?
[Here is the gist](https://gist.github.com/tj-devel709/03492abdf722192da23b84ae921f7e05) with the errors that marking mac for each of these interfaces show

Reply:
> It turns out that they are also in src/quicklookui.cs and are all decorated with mac attributes already.

Yes, that's the problem.

The fix is to remove these types from src/quicklookui.cs (if they're not exactly identical to the ones in src/quicklook.cs, those differences have to be copied to src/quicklook.cs), because this dual definition is certainly going to confuse us again in the future.

Reply:
@rolfbjarne Okay, I moved them over!